File: src/main/java/com/lessmarkup/userinterface/nodehandlers/user/UserProfileNodeHandler.java
package com.lessmarkup.userinterface.nodehandlers.user;

import com.google.inject.Inject;
import com.lessmarkup.interfaces.cache.DataCache;
import com.lessmarkup.userinterface.nodehandlers.common.TabPageNodeHandler;

public class UserProfileNodeHandler extends TabPageNodeHandler {

    @Inject
    public UserProfileNodeHandler(DataCache dataCache) {
        super(dataCache);
    }
}

File: src/main/scala/com/lessmarkup/userinterface/nodehandlers/user/UserProfileNodeHandler.scala
package com.lessmarkup.userinterface.nodehandlers.user

import com.google.inject.Inject
import com.lessmarkup.framework.nodehandlers.NodeHandlerConfiguration
import com.lessmarkup.interfaces.cache.DataCache
import com.lessmarkup.interfaces.structure.{NodeHandler, NodeHandlerFactory}
import com.lessmarkup.userinterface.nodehandlers.common.TabPageNodeHandler

class UserProfileNodeHandlerFactory @Inject() (dataCache: DataCache) extends NodeHandlerFactory {
  override def createNodeHandler(nodeHandlerConfiguration: NodeHandlerConfiguration, arguments: Any*): NodeHandler = {
    new UserProfileNodeHandler(dataCache, nodeHandlerConfiguration)
  }
}

class UserProfileNodeHandler(dataCache: DataCache, configuration: NodeHandlerConfiguration) extends TabPageNodeHandler(dataCache, configuration)